torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1753,22 +1794,74 @@ struct to_ir {
     };
     GatheredTypes gathered(typeParser_);
     gathered.gather(classinfo);
-    auto val = emitExpr(obj);
+    Value* val = emitExpr(obj);
     RefinementSet refinement;
-    if (gathered.types.size() == 1 &&
-        gathered.types.at(0)->isSubtypeOf(val->type()) &&
-        obj.kind() == TK_VAR) {
+
+    TORCH_CHECK(!gathered.types.empty(), "`isinstance` must be used with a "
```

Comment:
this is technically valid syntax... `isinstance(3, ())` it just always returns False